### PR TITLE
util: change some deprecate configs to system variables for OOM alarm

### LINKED ISSUE #38444: some configs used by OOM alarm are deprecated
## Enhancement
some configs used by OOM alarm are deprecated
deprecated config:
  - oom-action
  - mem-query-quota
  - server-memory-quota

### PR DESCRIPTION
**Pull request by mengxin9014**
### What problem does this PR solve?

Issue Number: close #38444

Problem Summary:
some configs are deprecate in OOM alarm and need to be replaced by system variables.

### What is changed and how it works?
change some deprecate configs to system variables for OOM alarm
|  Deprecated config   | Replace system variables  |
|  ----  | ----  |
| oom-action  | tidb_mem_oom_action |
| mem-quota-query  | tidb_mem_quota_query |
| server-memory-quota  | tidb_server_memory_limit |

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
change some deprecate configs to system variables for OOM alarm.
```

### CODE REVIEW

**Review by XuHuaiyu — APPROVED** [review]

File: util/memoryusagealarm/memoryusagealarm.go [hunk 1]
```diff
@@ -73,38 +73,42 @@ func (eqh *Handle) Run() {
...
+	lastUpdateVariableTime        time.Time
 	err                           error
 	baseRecordDir                 string
 	lastRecordDirName             []string
 	lastRecordMemUsed             uint64
 	memoryUsageAlarmRatio         float64
 	memoryUsageAlarmKeepRecordNum int64
-	serverMemoryQuota             uint64
-	isServerMemoryQuotaSet        bool
+	ServerMemoryLimit             uint64
```
Comment (wshwsh12) [inline util/memoryusagealarm/memoryusagealarm.go:83]:
```suggestion
	serverMemoryLimit             uint64
```

**Review by wshwsh12 — COMMENTED** [review]
Rest LGTM

File: util/memoryusagealarm/memoryusagealarm.go (hunk 1 above)
Comment (mengxin9014) [reply util/memoryusagealarm/memoryusagealarm.go]: > 

done.

File: util/memoryusagealarm/memoryusagealarm.go [hunk 2]
```diff
@@ -281,7 +275,10 @@ func (record *memoryUsageAlarm) getTop10SqlInfo(cmp func(i, j *util.ProcessInfo)
 	for i, info := range list {
 		buf.WriteString(fmt.Sprintf("SQL %v: \n", i))
 		fields := util.GenLogFields(record.lastCheckTime.Sub(info.Time), info, false)
-		fields = append(fields, zap.Int("analyze-version", info.OOMAlarmVariablesInfo.SessionAnalyzeVersion))
+		fields = append(fields, zap.String("tidb_mem_oom_action", variable.OOMAction.Load()))
+		fields = append(fields, zap.Uint64("tidb_server_memory_limit", memory.ServerMemoryLimit.Load()))
```
[the remark this answers is not in the document]
Comment (mengxin9014) [reply util/memoryusagealarm/memoryusagealarm.go]: > Can read these values only once

done.

**Review by wshwsh12 — APPROVED** [review]

**Pull request merged**